specification/compute/resource-manager/Microsoft.Compute/DiagnosticRP/preview/2024-03-01-preview/diagnostic.json

```diff
@@ -0,0 +1,665 @@
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/StorageConfigurationResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the get call failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Returns storageAccount value for an existing configuration entry": {
+            "$ref": "./examples/ValidateStorageConfiguration.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/diagnostics/spotPlacementRecommender/generate": {
+      "post": {
+        "tags": [
+          "spotPlacementRecommender"
+        ],
+        "operationId": "SpotPlacementRecommender_Post",
+        "description": "Generates placement scores for Spot VM skus.",
+        "parameters": [
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/LocationParameter"
+          },
+          {
+            "$ref": "../../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "spotPlacementRecommenderInput",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SpotPlacementRecommenderInput"
+            },
+            "description": "SpotPlacementRecommender object supplied in the body of the Post spot placement recommender operation."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/SpotPlacementRecommenderResponse"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Returns spot VM placement scores for given configurations.": {
+            "$ref": "./examples/SpotPlacementRecommender.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ComputeDiagnosticsMetadata": {
```

Comment:
>ComputeDiagnosticsMetadata [](http://example.com/codeflow?start=5&length=26)

Are you changing the name? Do you need to? List seemed to maek sense to me.

Reply:
Ah it seems like maybe it shouldn't have been a list response at all if it were following ARM conventions for a point GET.

Reply:
Resolving - name change to conform to ARM conventions